kms/api/CloudKmsTest/CloudKmsTest.cs

```diff
@@ -279,6 +294,124 @@ private bool AssertRoleAndMemberPresent(string response, string role, string mem
+            KeyRingName keyRingName = new KeyRingName(s_projectId, "global", keyRing);
+
+            Run("createKeyRing", keyRingName.ProjectId, keyRingName.LocationId, keyRingName.KeyRingId);
+
+            CryptoKeyVersionName version = BuildAsymmetricKey(
+                keyRingName, cryptoKey, CryptoKey.Types.CryptoKeyPurpose.AsymmetricSign,
+                CryptoKeyVersion.Types.CryptoKeyVersionAlgorithm.EcSignP256Sha256);
+
+            var signOutput = Run("asymmetricSign", version.ProjectId, version.LocationId, version.KeyRingId,
+                    version.CryptoKeyId, version.CryptoKeyVersionId, contentFile, signatureFile);
+            Assert.Equal(0, signOutput.ExitCode);
+
+            var verifyOutput = Run("asymmetricVerify", version.ProjectId, version.LocationId, version.KeyRingId,
+                    version.CryptoKeyId, version.CryptoKeyVersionId, contentFile, signatureFile);
+            Assert.Equal(0, verifyOutput.ExitCode);
+            Assert.Contains("verified: True", verifyOutput.Stdout);
+
+            // Delete signature file to clean up
+            File.Delete(signatureFile);
+        }
+
+        [Fact]
+        public void TestAsymmetricEncryptAndDecrypt()
+        {
+            string plaintextFile = @"../../../../test/data/test_file.txt";
+            string ciphertextFile = @"../../../../test/data/test_file_ciphertext.txt";
+            string recoveredPlaintextFile = @"../../../../test/data/test_file_recovered_plaintext.txt";
+            string timeStamp = $"-{DateTime.Now.ToString("yyyyMMddHHmmssfff")}";
+            string keyRing = $"testKeyRing{timeStamp}";
+            string cryptoKey = $"testCryptoKey{timeStamp}";
+            KeyRingName keyRingName = new KeyRingName(s_projectId, "global", keyRing);
+
+            Run("createKeyRing", keyRingName.ProjectId, keyRingName.LocationId, keyRingName.KeyRingId);
+
+            CryptoKeyVersionName version = BuildAsymmetricKey(
+                keyRingName, cryptoKey, CryptoKey.Types.CryptoKeyPurpose.AsymmetricDecrypt,
+                CryptoKeyVersion.Types.CryptoKeyVersionAlgorithm.RsaDecryptOaep2048Sha256);
+
+            var signOutput = Run("asymmetricEncrypt", version.ProjectId, version.LocationId, version.KeyRingId,
+                    version.CryptoKeyId, version.CryptoKeyVersionId, plaintextFile, ciphertextFile);
+            Assert.Equal(0, signOutput.ExitCode);
+            Assert.NotEqual(File.ReadAllBytes(plaintextFile), File.ReadAllBytes(ciphertextFile));
+
+            var verifyOutput = Run("asymmetricDecrypt", version.ProjectId, version.LocationId, version.KeyRingId,
+                    version.CryptoKeyId, version.CryptoKeyVersionId, ciphertextFile, recoveredPlaintextFile);
+            Assert.Equal(0, verifyOutput.ExitCode);
+            Assert.Equal(File.ReadAllBytes(plaintextFile), File.ReadAllBytes(recoveredPlaintextFile));
+
+            // Delete files to clean up
+            File.Delete(ciphertextFile);

```

Comment:
And also, delete what you created server side.

Reply:
As above

Reply:
As above.